File: content/chapters/compute/lab/solution/sleepy/sleepy_creator.py
# SPDX-License-Identifier: BSD-3-Clause

import subprocess
from sys import exit

NUM_SLEEPS = 10


def main():
    # TODO 1: Create 10 `sleep 1000` processes using `subprocess.Popen`
    # Use the documentation: https://docs.python.org/3/library/subprocess.html#subprocess.Popen
    procs = []
    for _ in range(NUM_SLEEPS):
        # Create new process and add it to the list of processes.
        p = subprocess.Popen(["sleep", "1000"])
        procs.append(p)

    # TODO 2: Make the current process wait for its child processes.
    for p in procs:
        p.wait()


if __name__ == "__main__":
    exit(main())

File: content/compute/lab/solution/sleepy/sleepy_creator.c
/* SPDX-License-Identifier: BSD-3-Clause */

#include <stdio.h>
#include <stdlib.h>

int main(void)
{
	/* TODO 1: Create one `sleep 1000` process using `system`. */
	system("sleep 1000");
	puts("Parent process ending.");
	return 0;
}
